### ISSUE TITLE
Run helmlint multiple times, with optional features enabled

### ISSUE THREAD

**Issue by jandubois** [cloudfoundry-incubator/kubecf#1156]
Right now we run `make helmlint` once, and it checks the helm chart with just the default settings for each optional feature.

Instead we should be running it once with each separate feature enabled (plus maybe 2 more for all features on, and all features off). This might not cover all possible expansions of the templates, but that would suffer from combinatorial explosion.

We have code in our `image_list.rb` script to enumerate all available feature, and expand the templates for various combinations (it used to test all combinations, but now only runs all features on, and all features off). This code could probably be extended to cover the helm linting use case as well.

**Comment by viccuad** [comment]
Could this be alleviated by adding a schema for the values.yaml?

**Comment by jandubois** [comment]
> Could this be alleviated by adding a JSON schema for the values.yaml?

@viccuad I don't think so. The JSON schema just validates the content of `values.yaml`. `helm lint` will render the templates with the current settings and makes sure the generated output is valid YAML. Since the templates will generate different content depending on the configured features, running `helm lint` against different combinations still makes sense to me.

However, this is still just a poor substitute for proper tests. Given the complexity of our templates, I think we should strive to have a broader suite of tests that generate charts for various configuration options, and then verify that the generated kube resource configurations match what we expect.

That will be a separate issue, that I plan on filing later this week. For one thing, we'll need a feature `cf-operator util interpolate` to apply all the generated ops files against the manifest. I'm going to file an issue for that one as well. 😄  We can get started by using `bosh int` for now, and there is code in `imagelist.rb` that does exactly that. But bosh doesn't support the additional `qcopy` and `qmove` operations that we added to quarks-operator. kubecf doesn't use them yet, but it will in the future, and then `bosh int` would no longer work.

Maybe if we can get started on proper tests sooner, then we don't need to bother with running `helm lint` for different configurations. I think that would be my preferred outcome.

PS and BTW, it just so happens that I wrote a schema for our `values.yaml` last week for my "multi-stack" work (still WIP): https://github.com/cloudfoundry-incubator/kubecf/blob/jandubois/1086-configurable-multi-stack-support/deploy/helm/kubecf/values.schema.yaml.

**Comment by mook-as** [comment]
> We can get started by using `bosh int` for now

I'm using [my own dumb wrapper](https://github.com/mook-as/go-patch-cmd) locally, but having it in `cf-operator` might be nice.  In the mean time that might be helpful.
